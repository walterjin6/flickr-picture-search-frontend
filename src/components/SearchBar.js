const SearchBar = () => {
 
  return (
    <div className='w-full'>

    </div>
  )
}

export default SearchBar
