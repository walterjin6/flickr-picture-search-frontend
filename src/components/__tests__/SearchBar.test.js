import SearchBar from '../SearchBar'
import { renderWithProviders } from '../../test/test-utils'

it('SearchBar renders correctly', () => {
  const { asFragment } = renderWithProviders(<SearchBar />)
  expect(asFragment()).toMatchSnapshot()
})



