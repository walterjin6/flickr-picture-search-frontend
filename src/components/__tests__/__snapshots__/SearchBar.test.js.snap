// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<DocumentFragment>
  <div
    class=" max-w-screen-2xl mx-auto"
  >
    <div
      class="mx-auto"
    >
      <div
        class="h-20 "
      >
        <header
          class="bg-white flex justify-between items-center text-center  gap-4 px-4 py-2 h-20 fixed top-0 left-0 right-0 z-50"
        >
          <svg
            class="w-1/12 h-full items-center text-center "
            height="800"
            overflow="visible"
            version="1.0"
            viewBox="-84.73395 -42.771 734.3609 256.626"
            width="1200"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M22.622 81.122H0V53.269h23.262V46.2C23.262 11.937 38.84 0 69.572 0c8.323 0 14.725 1.326 19.632 2.212L86.858 29.62c-3.2-1.105-6.402-1.77-12.164-1.77-9.816 0-13.658 7.074-13.658 18.349v7.072h27.316v27.852H61.036v87.307H22.622zM108.41 2.653h38.413v165.775H108.41zm71.278 2.652h38.414v29.177h-38.414zm0 47.965h38.414v115.157h-38.414zm153.869 32.933c-7.043-4.418-14.085-6.409-23.262-6.409-16.646 0-29.45 11.936-29.45 31.83 0 18.347 14.939 30.28 31.584 30.28 8.751 0 17.502-1.987 23.903-5.524l.853 29.618c-9.604 3.318-21.127 5.086-31.37 5.086-36.709 0-64.664-22.104-64.664-60.122 0-38.24 27.955-60.342 64.663-60.342 11.525 0 21.767 1.99 30.518 6.41zm58.048 15.475h.428l32.01-48.409h42.041l-38.626 52.827 41.828 62.332h-46.524l-30.73-57.025h-.427v57.025h-38.412V2.653h38.412z"
              fill="#0063db"
            />
            <path
              d="M562.97 85.098c-4.268-1.324-8.537-1.324-13.017-1.324-17.925 0-27.957 13.481-27.957 36.028v48.626h-38.412V53.27h34.997v21.22h.428c6.614-14.59 16.218-23.872 32.865-23.872 4.478 0 9.174.663 13.019 1.325z"
              fill="#ff0084"
            />
          </svg>
          <div
            class="w-full"
          >
            <label
              class="mb-2 text-sm font-medium text-gray-900 sr-only dark:text-gray-300"
            >
              Search
            </label>
            <div
              class="relative"
            >
              <div
                class="flex absolute inset-y-0 left-0 items-center pl-3 pointer-events-none"
              >
                <svg
                  aria-hidden="true"
                  class="w-5 h-5 text-gray-500 dark:text-gray-400"
                  fill="none"
                  stroke="currentColor"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </div>
              <input
                class="block p-4 pl-10 w-full text-base font-bold text-gray-900 bg-gray-50 rounded-lg border border-gray-300 focus:ring-blue-500 focus:border-blue-500 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
                id="default-search"
                placeholder=" Type  search  tag  keyword  here ..."
                type="search"
                value=""
              />
              <button
                class="text-white absolute right-2.5 bottom-2.5 bg-blue-700 hover:bg-blue-800 focus:ring-4 focus:outline-none focus:ring-blue-300 font-medium rounded-lg text-sm px-4 py-2 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800"
                type="submit"
              >
                Search
              </button>
            </div>
          </div>
          <div
            class="flex gap-1 text-xl text-gray-900 justify-center items-center "
          >
            <a
              href="/login"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                />
              </svg>
            </a>
          </div>
        </header>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SearchBar renders correctly 1`] = `
<DocumentFragment>
  <div
    class="w-full"
  >
    <label
      class="mb-2 text-sm font-medium text-gray-900 sr-only dark:text-gray-300"
    >
      Search
    </label>
    <div
      class="relative"
    >
      <div
        class="flex absolute inset-y-0 left-0 items-center pl-3 pointer-events-none"
      >
        <svg
          aria-hidden="true"
          class="w-5 h-5 text-gray-500 dark:text-gray-400"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </div>
      <input
        class="block p-4 pl-10 w-full text-base font-bold text-gray-900 bg-gray-50 rounded-lg border border-gray-300 focus:ring-blue-500 focus:border-blue-500 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
        id="default-search"
        placeholder=" Type  search  tag  keyword  here ..."
        type="search"
        value=""
      />
      <button
        class="text-white absolute right-2.5 bottom-2.5 bg-blue-700 hover:bg-blue-800 focus:ring-4 focus:outline-none focus:ring-blue-300 font-medium rounded-lg text-sm px-4 py-2 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800"
        type="submit"
      >
        Search
      </button>
    </div>
  </div>
</DocumentFragment>
`;
